library/src/engine.cxx

```diff
@@ -97,6 +101,72 @@ interactor& engine::getInteractor()
   return *this->Internals->Interactor;
 }
 
+//----------------------------------------------------------------------------
+void engine::loadPlugin(const std::string& path)
+{
+  // check if the plugin is a known static plugin
+  f3d::factory* factory = f3d::factory::instance();
+  factory::plugin_initializer_t init_plugin = factory->getStaticInitializer(path);
+
+  if (init_plugin == nullptr)
+  {
+    // try if the path is a library full path
+    vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(path);
+    if (!handle)
+    {
+      std::string fullPath;
+
+      // Right now, plugins should be located in the same folder than f3d.exe on Windows
+      // On Linux/macOS, when we are not using a full path, so we rely on LD_LIBRARY_PATH
+#ifdef _WIN32
+      std::vector<wchar_t> pathBuf(40000);
+      if (GetModuleFileNameW(
+            GetModuleHandle(nullptr), pathBuf.data(), static_cast<DWORD>(pathBuf.size())))
+      {
+        std::string progPath =
+          vtksys::SystemTools::GetProgramPath(vtksys::Encoding::ToNarrow(pathBuf.data()));
+        fullPath = vtksys::SystemTools::ConvertToWindowsOutputPath(progPath) + "\\";
+      }
+#endif
+
+      // construct the library file name from the plugin name
+      fullPath += vtksys::DynamicLoader::LibPrefix();
+      fullPath += "f3d-plugin-";
+      fullPath += path;
```

Comment:
how, do we want to put the `f3d-plugin` part or let the suer do it ?

Reply:
It's too redundant in my opinion. I don't think it's great to let the user write `--load-plugins=f3d-plugin-xxx,f3d-plugin-yyy,f3d-plugin-zzz`

Reply:
ok, needs to be clearly documented though

Reply:
It's already documented in the header: 

```cpp
  /**
   * Load a plugin.
   * Supports full path, relative path, and plugin name.
   * On Linux and macOS, uses LD_LIBRARY_PATH to find the plugin.
   * On Windows, the plugin should be located in the same folder as the executable.
   */
  static void loadPlugin(const std::string& path);
```

I'll add this in the final documentation when your PR will be merged to avoid conflicts

Reply:
ok, lets move my PR forward then